src/bin/add/args.rs

```diff
@@ -85,17 +85,11 @@ pub struct Args {
     pub optional: bool,
 
     /// Path to the manifest to add a dependency to.
-    #[structopt(long = "manifest-path", value_name = "path", conflicts_with = "pkgid")]
-    pub manifest_path: Option<PathBuf>,
+    #[structopt(flatten)]
+    pub manifest: clap_cargo::Manifest,
```

Comment:
Could you focus the PR, or at least the commits, to only porting to one clap_cargo feature at a time?  It'll make it easier to see the impact.

Reply:
I'll implement `Workspace`, `Features` & `Manifest` in 3 separate commits, right?

Reply:
That works!

Separate PRs are nice because
- This project is configured for squash-merge only and separate PRs will preserve the distinction
- If there are easier / harder parts, we can get the easier parts through more quickly, allowing developing more incrementally rather than everything having to be ready at once

Separate commits are nice because
- Its easier to iterate on all the parts without any blockage due to PR reviews.

Reply:
@epage I'll work in this PR for now, maybe later I'll change my mind & file another 2 PRs.